test/classes/initializers/compilerGenerated/arrayField-sync.good

```diff
@@ -1,2 +1,3 @@
 arrayField-sync.chpl:2: warning: compiler generated default initializers for classes with 'sync' fields are deprecated, please supply an 'init' method
+arrayField-sync.chpl:3: warning: returning a sync by value is deprecated
```

Comment:
I was curious about the impacts on existing tests, so was just browing the PR from that perspective.  For this whole directory of changes, I could imagine the new warning to be confusing because the user doesn't seem to be returning a sync (i.e., I'm guessing it's referring either to the compiler-generated initializer, or possibly to the compiler-generated setter/getter routine(s)—is that right?)

Maybe asking the same question a different way: If I were to add an initializer that resolved the warning on line 2 here, would the line 3 warning go away as well or would it persist?

Reply:
In this case, the deprecation is for the compiler generated "default actual function". Adding a user initializer does make both warnings go away. Would it make more sense to just skip warning for this case?

Reply:
Thanks!  If that were easy to do, I'd consider it a benefit because I think the current warning is likely to cause users to say "I'm not returning a sync... what are you talking about?"  That said, I'm reassured that it's due to the initializer because I can imagine leaning on the rationale that "once you get a warning or error, subsequent errors/warnings may cascade from it and be of diminishing value."  That said, it'd obviously be better for us not to generate low-value warnings if we're able to.

Are these default actual functions used for anything other than compiler-generated initializers?  If not it seems like it might be fairly easy to add to the list of exceptions you've already got (?).

Reply:
It should be easy to skip the warning here, I only see these getting adding when creating default initializers. There is something for it in virtualDispatch that uses these defaulted actual functions, but I wasn't able to write a test case that triggers warnings in those cases. 

With a only a cursory understanding of what the compiler is doing exactly in these cases, I think I can just add the exception.

Regardless, these are compiler generated functions that are not called directly by users anyways (and will be deleted for sync/single when we remove the feature).

Reply:
I couldn't imagine other places that such routines would be necessary and can't think of why they would be in the virtual dispatch logic.  I think applying the big hammer to the helper routine is the right way to go, esp. given your inability to write a test that would trigger the virtual dispatch case.  In part because it's easier and in part because it may help us run into such cases later, if they really exist (well, and we run into them before we remove the deprecations altogether).

Thanks!